st2common/st2common/services/sensor_watcher.py

```diff
@@ -95,9 +95,18 @@ def start(self):
             self.connection.release()
 
     def stop(self):
+        LOG.debug('Shutting down sensor watcher.')
         try:
             if self._updates_thread:
                 self._updates_thread = eventlet.kill(self._updates_thread)
+
+            if self.connection:
+                channel = self.connection.channel()
```

Comment:
But the queue should get re-created when connection is successfully re-established, right?

If not, we can / should utilize `declare=` functionality I mentioned in #3648